components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java

```diff
@@ -642,6 +655,26 @@ private static double getReCaptchaThreshold(Properties properties) throws Number
         return Double.parseDouble(threshold);
     }
 
+    /**
+     * Method to get the warn threshold value used by reCAPTCHA v3.
+     *
+     * @param properties Properties.
+     * @return Warn threshold value set by the user or the default warn threshold.
+     * @throws java.lang.NumberFormatException Error while parsing the threshold value into double.
+     */
+    private static double getReCaptchaWarnThreshold(Properties properties) throws NumberFormatException {
+
+        String warnThreshold = properties.getProperty(CaptchaConstants.RE_CAPTCHA_WARN_SCORE_THRESHOLD);
+        if (StringUtils.isBlank(warnThreshold)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error parsing recaptcha.threshold.warn from config. Hence using the default value : " +
+                        CaptchaConstants.CAPTCHA_V3_DEFAULT_WARN_THRESHOLD);
+            }
+            return CaptchaConstants.CAPTCHA_V3_DEFAULT_WARN_THRESHOLD;
+        }
+        return Double.parseDouble(warnThreshold);
```

Comment:
Can we catch the number format exception here print a warn log saying the value is not valid so going with default?

Reply:
Exception handling here also has been done similar to recaptcha score threshold 
https://github.com/wso2-extensions/identity-governance/blob/master/components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java#L632-L643

Should I change it or keep this to maintain consistency?

Reply:
+1 to print a warn and assing the default value

Reply:
Done with [7c693bd](https://github.com/wso2-extensions/identity-governance/pull/768/commits/7c693bda8d996fc7b4fb5fb7f201709b840aaa37)